compiler/rustc_passes/src/lang_items.rs

```diff
@@ -31,151 +31,181 @@ pub(crate) enum Duplicate {
     CrateDepends,
 }
 
-struct LanguageItemCollector<'tcx> {
+struct LanguageItemCollector<'ast, 'tcx> {
     items: LanguageItems,
     tcx: TyCtxt<'tcx>,
+    resolver: &'ast ResolverAstLowering,
+    item_spans: FxHashMap<DefId, Span>,
```

Comment:
I wonder if we should https://github.com/rust-lang/rust/issues/118552 to avoid having to maintain such map. Could you add a FIXME with the issue so we can retrieve the question?